lib/core/src/Cardano/Wallet.hs

```diff
@@ -1040,14 +1055,29 @@ restoreBlocks ctx tr wid blocks nodeTip = db & \DBLayer{..} -> mapExceptT atomic
                 -- anyway.
                 cfg = (defaultSparseCheckpointsConfig epochStability) { edgeSize = 0 }
 
-    forM_ (NE.init cps) $ \cp' -> do
-        let (Quantity h) = currentTip cp' ^. #blockHeight
-        when (fromIntegral h `elem` unstable) $ do
-            liftIO $ logCheckpoint cp'
-            putCheckpoint wid cp'
+        getBlockHeight cp = fromIntegral $
+            cp ^. #currentTip . #blockHeight . #getQuantity
+        willKeep cp = getBlockHeight cp `Set.member` unstable
+        cpsKeep = filter willKeep (NE.init cps) <> [NE.last cps]
 
-    liftIO $ logCheckpoint (NE.last cps)
-    putCheckpoint wid (NE.last cps)
+        -- NOTE: We have to update the 'Prologue' as well,
+        -- as it can contain addresses for pending transactions,
```

Comment:
well, the question is whether Prologue needs to have addresses of pending txs. It does not sounds like static data. Nevertheless, pending txs needs o be taken care when rollbacks happen